### PR TITLE
refactor(dashmate): status command

### PR DESCRIPTION
**Pull request by pshenmic**
## Issue being fixed or feature implemented

Refactored plenty of legacy status command code: removed all code duplication, separated data acquiring from the view, optimized json structures, added first dashmate unit/integration tests.

More info at #556

## What was done?

* broke down all status data code into "scopes"
* removed all relevant code duplication
* added unit and integration tests for feature functionality

## How Has This Been Tested?

Locally on arm64 laptop + tests

## Breaking Changes

* `dashmate status` and `dashmate group status` commands now output new plain and json structures

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dashmate/src/status/statusProviderFactory.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const getCoreScopeFactory = require('./scopes/core');
```
Comment (shumkov) [inline packages/dashmate/src/status/statusProviderFactory.js:1]: Why do you need this if we have already DI container?

File: packages/dashmate/src/status/scopes/services.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+const ContainerIsNotPresentError = require('../../docker/errors/ContainerIsNotPresentError');
+
+module.exports = (dockerCompose) => async (config) => {
```
Comment (shumkov) [inline packages/dashmate/src/status/scopes/services.js:3]: We create factories a little bit differently. Please look at JS Drive:


servicesFactory.js
```js
/**
 * @returns {services}
 */
function servicesFactory(dep) {
  /**
   * @typedef services
   */ 
  function services() {
    ...
  }

  return services;
}

module.exports = servicesFactory;
```

File: packages/dashmate/test/status/mocks/mnSyncMock.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (shumkov) [inline packages/dashmate/test/status/mocks/mnSyncMock.json:1]: We usually have "global" mocks dir, like `packages/dashmate/test/mocks/status/mnSyncMock.json` so other tests can use them not just status.

File: packages/dashmate/src/status/statusProviderFactory.js (hunk 1 above)
Comment (pshenmic) [reply packages/dashmate/src/status/statusProviderFactory.js:1]: This function does not participate in DI

**Comment by jaboski** [comment]
I'm having random issues for e.g. group status although docker ps show everything is running. It shows up randomly. Have you seen that before?

`FetchError: request to http://localhost:26857/net_info failed, reason: socket hang up
    at ClientRequest.<anonymous> (/home/adam/testing/platform/.yarn/__virtual__/node-fetch-virtual-d3846f8e12/0/cache/node-fetch-npm-2.6.7-777aa2a6df-8d816ffd1e.zip/node_modules/node-fetch/lib/index.js:1491:11)
    at ClientRequest.emit (node:events:520:28)
    at Socket.socketOnEnd (node:_http_client:466:9)
    at Socket.emit (node:events:532:35)
    at endReadableNT (node:internal/streams/readable:1346:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET'
}`

**Comment by pshenmic** [comment]
I saw that when tenderdash were constantly restarting on my system due incorrect docker images. `dashmate update` resolved me that

File: packages/dashmate/src/commands/status/index.js [hunk 4]
```diff
@@ -2,22 +2,101 @@ const { Flags } = require('@oclif/core');
...
     args,
     flags,
-    outputStatusOverview,
+    dockerCompose,
+    getOverviewScope,
     config,
   ) {
-    await outputStatusOverview(config, flags.format);
+    if (!(await dockerCompose.isServiceRunning(config.toEnvs()))) {
+      // eslint-disable-next-line no-console
```
Comment (shumkov) [inline packages/dashmate/src/commands/status/index.js:27]: Just throw an error

File: packages/dashmate/src/commands/status/platform.js [hunk 5]
```diff
@@ -26,186 +25,73 @@ class PlatformStatusCommand extends ConfigBaseCommand {
     dockerCompose,
     createRpcClient,
     config,
+    getPlatformScope,
   ) {
     if (config.get('network') === 'mainnet') {
       // eslint-disable-next-line no-console
       console.log('Platform is not supported on mainnet yet!');
```
Comment (shumkov) [inline packages/dashmate/src/commands/status/platform.js:32]: throw an error

File: packages/dashmate/src/commands/status/platform.js [hunk 6]
```diff
@@ -26,186 +25,73 @@ class PlatformStatusCommand extends ConfigBaseCommand {
...
-      },
-    } = await coreService.getRpcClient().mnsync('status');
+    // Collect platform data
+    const scope = await getPlatformScope(config);
 
     // Collecting platform data fails if Tenderdash is waiting for core to sync
-    if (coreIsSynced === false) {
+    if (!scope.coreIsSynced) {
       // eslint-disable-next-line no-console
       console.log('Platform status is not available until core sync is complete!');
```
Comment (shumkov) [inline packages/dashmate/src/commands/status/platform.js:46]: throw an error

File: packages/dashmate/src/core/getPaymentQueuePosition.js [hunk 7]
```diff
@@ -1,18 +1,18 @@
-function getPaymentQueuePosition(masternodeState, masternodeEnabledCount, coreBlocks) {
+function getPaymentQueuePosition(dmnState, masternodeEnabledCount, coreBlocks) {
```
Comment (shumkov) [inline packages/dashmate/src/core/getPaymentQueuePosition.js:1]: `calculatePaymentQueuePosition `

File: packages/dashmate/src/printers/printObject.js [hunk 8]
```diff
@@ -20,10 +20,15 @@ function printObject(object, format) {
...
     }
     case OUTPUT_FORMATS.JSON: {
-      const cleanObject = {};
-      Object.keys(object).forEach((key) => {
-        cleanObject[key] = stripAnsi(object[key]);
-      });
+      let cleanObject = {};
+      if (Array.isArray(object)) {
+        cleanObject = [];
+        object.forEach((e) => cleanObject.push(stripAnsi(e)));
```
Comment (shumkov) [inline packages/dashmate/src/printers/printObject.js:26]: Do we still need strip ANSI?

File: packages/dashmate/src/status/scopes/host.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+   * @returns {Promise<Object>}
+   */
+  async function getHostScope() {
+    return ({
+      hostname: os.hostname(),
+      uptime: prettyMs(os.uptime() * 1000),
+      platform: os.platform(),
+      arch: os.arch(),
+      username: os.userInfo().username,
+      diskFree: 0, // Waiting for feature: https://github.com/nodejs/node/pull/31351
```
Comment (shumkov) [inline packages/dashmate/src/status/scopes/host.js:23]: Better don't show 0.

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/src/printers/printArrayOfObjects.js [hunk 10]
```diff
@@ -31,7 +30,7 @@ function printArrayofObjects(array, format) {
       array.forEach((outputRow, i) => {
```
Comment (shumkov) [inline packages/dashmate/src/printers/printArrayOfObjects.js:30]: Do you still need this?

File: packages/dashmate/src/printers/printObject.js [hunk 11]
```diff
@@ -20,10 +19,15 @@ function printObject(object, format) {
       break;
     }
     case OUTPUT_FORMATS.JSON: {
-      const cleanObject = {};
-      Object.keys(object).forEach((key) => {
-        cleanObject[key] = stripAnsi(object[key]);
-      });
+      let cleanObject = {};
```
Comment (shumkov) [inline packages/dashmate/src/printers/printObject.js:22]: Do you still need this?

**Review by shumkov — CHANGES_REQUESTED** [review]

**Pull request merged**